### PR TITLE
Add unit tests for hangup / reject

### PR DESCRIPTION
**Pull request by dbkr**
Fixes https://github.com/vector-im/element-call/issues/537

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by SimonBrandner — APPROVED** [review]

**Comment by SimonBrandner** [comment]
Except the tests actually don't seem to be passing :/

File: src/webrtc/call.ts [hunk 1]
```diff
@@ -1569,13 +1569,20 @@ export class MatrixCall extends TypedEventEmitter<CallEvent, CallEventHandlerMap
 
         try {
             await this.peerConn.setLocalDescription(answer);
+
+            // make sure we're still going
+            if (this.callHasEnded()) return;
+
```
Comment (SimonBrandner) [inline src/webrtc/call.ts:1575]: I wonder if this should be in `setState()`

File: src/webrtc/call.ts [hunk 2]
```diff
@@ -1569,13 +1569,20 @@ export class MatrixCall extends TypedEventEmitter<CallEvent, CallEventHandlerMap
...
             this.setState(CallState.Connecting);
 
             // Allow a short time for initial candidates to be gathered
             await new Promise(resolve => {
                 setTimeout(resolve, 200);
             });
 
+            // make sure the call hasn't ended before we continue
+            if (this.callHasEnded()) return;
+
```
Comment (SimonBrandner) [inline src/webrtc/call.ts:1585]: I wonder if this should be in `sendAnswer()`

File: src/webrtc/call.ts (hunk 1 above)
Comment (SimonBrandner) [reply src/webrtc/call.ts:1575]: I should go sleep, anyway...

**Review by SimonBrandner — APPROVED** [review]

File: src/webrtc/call.ts (hunk 1 above)
Comment (dbkr) [reply src/webrtc/call.ts:1575]: Yeah, I did wonder about this and it seems quite elegant to basically just disallow state transitions out of 'ended', at the price of having to wrap all our setState calls in a try/catch (or some way to make the caller stop what it's doing).

File: src/webrtc/call.ts (hunk 2 above)
Comment (dbkr) [reply src/webrtc/call.ts:1585]: Possibly - I went for here as it's straight after it comes back from await-ing.

File: src/webrtc/call.ts (hunk 1 above)
Comment (SimonBrandner) [reply src/webrtc/call.ts:1575]: Yeah, so what I meant by _I should go to sleep, anyway..._ is that there is no simple way to do this, so I think this is actually fine (same for the other comment)

**Pull request merged**